homeassistant/components/ecobee/climate.py

```diff
@@ -414,10 +419,18 @@ def target_temperature_step(self) -> float:
     def has_humidifier_control(self):
         """Return true if humidifier connected to thermostat and set to manual/on mode."""
         return (
```

Comment:
You could write this similar to https://github.com/home-assistant/core/pull/86100/files#r1083346624

Reply:
Changed to 
```
        return (
            self.thermostat["settings"].get("humidifierMode") == HUMIDIFIER_MANUAL_MODE
        )

```

Reply:
Changed again, as I noticed I did it slightly wrong.

As well, the old method(both previous PR's version and current core's version) was only check to see if "hasHumidifier" existed, not if it was actually true or not. Changing this meant I had to change a few of the test to reflect that fact.
```
        return (
            bool(self.thermostat["settings"].get("hasHumidifier"))
            and self.thermostat["settings"].get("humidifierMode")
            == HUMIDIFIER_MANUAL_MODE
        )
```

Reply:
I'd probably write this whole block as 
```python
    @property
    def settings(self) -> dict[str, Any]:
        """Return the settings of the thermostat."""
        return self.thermostat["settings"]

    @property
    def has_humidifier_control(self) -> bool:
        """Return true if humidifier connected to thermostat and set to manual/on mode."""
        return (
            bool(self.settings.get("hasHumidifier"))
            and self.settings.get("humidifierMode") == HUMIDIFIER_MANUAL_MODE
        )

    @property
    def has_aux_heat(self) -> bool:
        """Return true if the ecobee has a heat pump."""
        return bool(self.settings.get(HAS_HEAT_PUMP))
```